From: Maxim Kuvyrkov <maxim@codesourcery.com>
Date: Wed, 10 Feb 2010 15:24:21 +0000 (-0800)
Subject: Add m68k TLS relocations
X-Git-Tag: glibc-2.12~194
X-Git-Url: http://sourceware.org/git/?p=glibc.git;a=commitdiff_plain;h=bbabf1f73f611b9101b42c3390a9ddd53e6dd7d4

Add m68k TLS relocations
---

#diff --git a/ChangeLog b/ChangeLog
#index 5ab2cbc..cea1089 100644
#--- a/ChangeLog
#+++ b/ChangeLog
#@@ -1,3 +1,7 @@
#+2010-02-09  Maxim Kuvyrkov  <maxim@codesourcery.com>
#+
#+	* elf/elf.h: Define m68k TLS relocations.
#+
# 2010-02-10  Luis Machado  <luisgpm@br.ibm.com>
# 
#	 * sysdeps/powerpc/powerpc64/power7/Implies: Removed.
diff --git a/elf/elf.h b/elf/elf.h
index 8af7c17..8b1ee47 100644
--- a/elf/elf.h
+++ b/elf/elf.h
@@ -1123,8 +1123,29 @@ typedef struct
 #define R_68K_GLOB_DAT	20		/* Create GOT entry */
 #define R_68K_JMP_SLOT	21		/* Create PLT entry */
 #define R_68K_RELATIVE	22		/* Adjust by program base */
+#define R_68K_TLS_GD32      25          /* 32 bit GOT offset for GD */
+#define R_68K_TLS_GD16      26          /* 16 bit GOT offset for GD */
+#define R_68K_TLS_GD8       27          /* 8 bit GOT offset for GD */
+#define R_68K_TLS_LDM32     28          /* 32 bit GOT offset for LDM */
+#define R_68K_TLS_LDM16     29          /* 16 bit GOT offset for LDM */
+#define R_68K_TLS_LDM8      30          /* 8 bit GOT offset for LDM */
+#define R_68K_TLS_LDO32     31          /* 32 bit module-relative offset */
+#define R_68K_TLS_LDO16     32          /* 16 bit module-relative offset */
+#define R_68K_TLS_LDO8      33          /* 8 bit module-relative offset */
+#define R_68K_TLS_IE32      34          /* 32 bit GOT offset for IE */
+#define R_68K_TLS_IE16      35          /* 16 bit GOT offset for IE */
+#define R_68K_TLS_IE8       36          /* 8 bit GOT offset for IE */
+#define R_68K_TLS_LE32      37          /* 32 bit offset relative to
+					   static TLS block */
+#define R_68K_TLS_LE16      38          /* 16 bit offset relative to
+					   static TLS block */
+#define R_68K_TLS_LE8       39          /* 8 bit offset relative to
+					   static TLS block */
+#define R_68K_TLS_DTPMOD32  40          /* 32 bit module number */
+#define R_68K_TLS_DTPREL32  41          /* 32 bit module-relative offset */
+#define R_68K_TLS_TPREL32   42          /* 32 bit TP-relative offset */
 /* Keep this the last entry.  */
-#define R_68K_NUM	23
+#define R_68K_NUM	43
 
 /* Intel 80386 specific definitions.  */
 
